foreman/package.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+function frmn_main () {
+  [[ -z "$WORKERNAME" ]] && {
+    echo "WORKERNAME not set"
+    return 1
+  } 
+  [[ -z "$FOREMAN_CLIENT_ID" ]] && {
+    echo "FOREMAN_CLIENT_ID not set"
+    return 1
+  } 
+  [[ -z "$FOREMAN_API_KEY" ]] && {
+    echo "FOREMAN_API_KEY not set"
+    return 1
+  } 
+  return 0
+}
+
+function frmn_create_pickaxe() {
+    local latest_version=$(frmn_get_latest_pickaxe_version)
+
+    # Create a new pickaxe instance in Foreman for this rig
+    local pickaxe_json=$(jq -n --arg client $FOREMAN_CLIENT_ID \
+                            --arg plabel "$WORKERNAME" \
+                            --arg version "$latest_version" '. | 
+                                {
+                                    client: $client|tonumber,
+                                    label: $plabel,
+                                    version: $version
+                                }')
+    echo $(curl \
+            -s \
+            -X POST \
+            -H "Content-Type: application/json" \
+            -H "Authorization: Token $FOREMAN_API_KEY" \
+            -d "$pickaxe_json" \
+            https://dashboard.foreman.mn/api/pickaxe/$FOREMAN_CLIENT_ID | \
+            jq -r '.key')
+}
+
+function frmn_get_current_chisel_version() {
+    if [ -d $NVOC/foreman/chisel ]; then
+        echo $(unzip \
+                -q \
+                -c $NVOC/foreman/chisel/lib/chisel-*.jar META-INF/MANIFEST.MF | \
```

Comment:
chisel/pickaxe binaries should not be installed within the active nvoc instance folder to allow sharing of binaries across different nvoc instances/versions and avoiding adding external files to the local git repo clone. ~/.foreman/ could be a good candidate location for this purpose.

Reply:
I can change where it gets installed to.

Is there a reason why you’d like for it to be a hidden folder? Just want to make sure that wasn’t a typo before I make that change.

Reply:
It was not a typo. It is just my default policy whenever I add some folder to the user home which is not user content (documents, photos, etc..). Please, do as you think it's best, it does not make any real difference for me.